### ISSUE TITLE
java.lang.RuntimeException · Illegal callback invocation from native module. This callback type only permits a single invocation from native code.

### ISSUE THREAD

**Issue by SudoPlz** [Microsoft/react-native-code-push#1259]
Received the following bug in Bugsnag from one of the devices in production:

    java.lang.RuntimeException Illegal callback invocation from native module. This callback type only permits a single invocation from native code. 
        CallbackImpl.java:30 com.facebook.react.bridge.CallbackImpl.invoke
        CodePushDialog.java:66 com.microsoft.codepush.react.CodePushDialog$2.onClick
        AlertController.java:175 com.android.internal.app.AlertController$ButtonHandler.handleMessage
        Handler.java:102 android.os.Handler.dispatchMessage
        Looper.java:154 android.os.Looper.loop
        MessageQueueThreadImpl.java:194 com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run
        Thread.java:762 java.lang.Thread.run

Physical device: Android 7.0 (API 24)
"react-native": "0.53.3",
"react-native-code-push": "5.2.2",

**Comment by buptkang** [comment]
Hey, have you ever by chance seen this similar [issue](https://github.com/facebook/react-native/issues/13595) from react-native repo?

**Comment by alexandergoncharov** [comment]
Hi @SudoPlz ,
Sorry for long delay.

There isn't any fix implementation for this as this issue in React-native was closed as inactive.

Could you please share any more logs of code example which can help us to reproduce issue?

Thanks,
Alexander

**Comment by SudoPlz** [comment]
I don't know much about this issue as the user wasn't doing anything when this happened, and it happened on a production device for which we received a log.

Here's more info in case it helps:

```
apiLevel: 25
batteryLevel: 0.63
brand: motorola
charging: false
cpuAbi: [
  "armeabi-v7a",
  "armeabi"
]
dpi: 272
freeDisk: 10766757888
freeMemory: 190145816
id: d96d5afc-8675-4707-86b6-aa8c8512fb6d
jailbroken: false
locale: en_US
locationStatus: allowed
manufacturer: motorola
model: Moto E (4) Plus
networkAccess: cellular
orientation: portrait
osBuild: NPRS26.58-45-6
osName: android
osVersion: 7.1.1
screenDensity: 1.7
screenResolution: 1198x720
time: 2018-05-11T14:41:53.000Z
totalMemory: 201326592
```

The only other piece of info I can attach is the threads at the time of the crash all-though I doubt that will help.

What I can tell you is, that crash probably took place, when the user was asked wether they want to update to the latest version or not, and the clicked on the codepush dialog.

**Comment by SudoPlz** [comment]
Hi again @alexandergoncharov nope, I haven't tried reproducing that, and I don't think I can reproduce that at will on my device (since I've been clicking that dialog for a long time now without crashes) and I don't have a Moto E (4) Plus on 7.1.1 to test it with the same device as that user.

**Comment by NickToropov** [comment]
Hi @SudoPlz, 
I think I was able to reproduce the issue and fix it. Please take a look at [PR](https://github.com/Microsoft/react-native-code-push/pull/1289)

**Comment by SudoPlz** [comment]
@NickToropov the PR looks good my friend, can't wait to get that merged.

**Comment by NickToropov** [comment]
Hi @SudoPlz, 

we've released a new plugin version - 5.3.4 with fix for the issue.
Please, update your application with new plugin.

**Comment by SudoPlz** [comment]
I'll close this for now in hopes that it's resolved. If it turns out to persist I'll re-open.

Thanks @NickToropov
